Doc/library/exceptions.rst

```diff
@@ -851,6 +851,68 @@ The following exceptions are used as warning categories; see the
    .. versionadded:: 3.2
 
 
+Exception groups
+----------------
+
+The following are used when it is necessary to raise multiple unrelated
+exceptions. They are part of the exception hierarchy so they can be
+handled with :keyword:`except` like all other exceptions. In addition,
+they are recognised by :keyword:`except*<except_star>`, which matches
+their subgroups based on the types of the contained exceptions.
+
+.. exception:: ExceptionGroup(msg, excs)
+.. exception:: BaseExceptionGroup(msg, excs)
+
+   Both of these exception types wrap the exceptions in the sequence ``excs``.
+   The ``msg`` parameter must be a string. The difference between the two
+   classes is that :exc:`BaseExceptionGroup` extends :exc:`BaseException` and
+   it can wrap any exception, while :exc:`ExceptionGroup` extends :exc:`Exception`
+   and it can only wrap subclasses of :exc:`Exception`. This design is so that
+   ``except Exception`` catches an :exc:`ExceptionGroup` but not
+   :exc:`BaseExceptionGroup`.
+
+   It is usually not necessary for a program to explicitly create a
+   :exc:`BaseExceptionGroup`, because the :exc:`ExceptionGroup` constructor
```

Comment:
This documentation is more precise - ``BaseException.__new__`` acts as this factory method.  Shall I update the PEP?

Reply:
> This documentation is more precise - `BaseException.__new__` acts as this factory method. Shall I update the PEP?

I'm uncomfortable with that, because the PEP that the SC approved (at least as I interpreted it) was a separate factory function.  So I think if you want to go with `BaseException.__new__`, that will require some discussion with the new SC.  

FWIW, I think a separate factory method would be less magical.  Is there a particular reason why you favor `BaseException.__new__` over a more explicit, separate factory method?

Reply:
@gvanrossum @1st1

Reply:
Aren't you all slightly confused? The factory function is `BaseExceptionGroup.__new__`, not `BaseException.__new__`. To me having this be built into the `BaseExceptionGroup` constructor seems fine. But I don't feel very strongly either way, so if there's a procedural issue with interpreting the PEP this way we can invent a factory function. (Does it have to be exposed to Python, or can it be a private C function?)

Reply:
Yes, sorry that was a typo. BaseExceptionGroup of course.

Reply:
I think it does need to be in python, the purpose of the factory function is to make it possible for you to not have to worry about whether you need an ExceptionGroup or a BaseExceptionGroup.

So to ask the SC how to resolve this, is that an issue on the SC tracker, or an email to python-dev or both?

Reply:
I'd ask it on the SC public tracker, at least to get their (I say this as a former member 😄 ) read on it given the language in the approved PEP.  Depending on what they say, then a follow up on python-dev may be useful.  For me, I just think it's a little jarring that `BaseExceptionGroup(...)` might return an instance of a subclass.

Reply:
> For me, I just think it's a little jarring that `BaseExceptionGroup(...)` might return an instance of a subclass.

But it's still an instance of `BaseExceptionGroup`! This kind of thing has been supported ever since we added `__new__`. And this kind of microscopic deviation from the PEP once the implementation is under way is why we don't wait until the last minute getting PEPs approved. IIRC doing this in `__new__` was always the intention, and I'd much rather fix this in the PEP than in the code.

Reply:
(In Irit's original implementation branch, which was functional long before the PEP was accepted, this is also done in `BaseExceptionGroup.__new__`. That shows that the PEP authors intended it to be that way, and we just used a somewhat awkward wording in the PEP.)

Reply:
Okay, that's good to know that this is documenting the original intention.  If the wording in the PEP was just a bit inelegant in expressing this, then it should be an easy fix.  I'd still recommend opening an issue on the SC tracker.  I'm mindful of features changing between the time the PEP is approved and when the implementation lands.  There have been other cases where the changes weren't so innocuous, so I think it's better to err on the side of transparency.  In this case, the SC should be able to just approve and close it quickly.  While it's inevitable that such changes will happen, I think the SC should at least be in the loop during the release in which the feature is being added.  After that, we have backward compatibility constraints and other ways of being careful (and PEPs are a snapshot in time, not a living documenting being continually updated to track a feature over of the life of ~~Brian~~ er, Python).

Reply:
My apologies - I got it wrong in this documentation. The BaseExceptionGroup constructor is indeed the factory function, and it will return an ExceptionGroup if it case. But the ExceptionGroup constructor will not allow you to wrap a BaseException, it raises TypeError. I will push a fix now.